rds_to_s3/etl_lambda.py

```diff
@@ -93,39 +75,41 @@ def s3_connection() -> boto3.client:
     """Function connects to S3 and provides client to interact with it."""
     return boto3.client(
         "s3",
-        aws_access_key_id=ENV["AWS_ACCESS_KEY_ID"],
-        aws_secret_access_key=ENV["AWS_SECRET_ACCESS_KEY"]
+        aws_access_key_id=ENV["ACCESS_KEY_ID"],
+        aws_secret_access_key=ENV["SECRET_ACCESS_KEY"]
     )
 
 
-def download_csv_from_s3(bucket_name: str, file_name: str) -> pd.DataFrame:
-    """Downloads the current archive csv from S3"""
+def download_csv_from_s3(bucket_name: str, s3_file_path: str, file_name: str) -> pd.DataFrame:
+    """Downloads the files from the s3 bucket"""
+    tmp_file_name = f"/tmp/{file_name}"
     s3 = s3_connection()
     try:
-        s3.download_file(bucket_name, file_name, file_name)
-        logging.info("Downloaded %s from S3.", file_name)
-        return pd.read_csv(file_name)
+        s3.download_file(bucket_name, s3_file_path, tmp_file_name)
+        logging.info("Downloaded %s from S3.", tmp_file_name)
+        return pd.read_csv(tmp_file_name)
     except FileNotFoundError as fe:
-        logging.warning("File %s not found locally: %s", file_name, fe)
+        logging.warning("File %s not found locally: %s", tmp_file_name, fe)
         return pd.DataFrame()
     except ClientError as ce:
         logging.warning(
-            "Failed to download %s from S3 (AWS Client Error): %s", file_name, ce)
+            "Failed to download %s from S3 (AWS Client Error): %s", tmp_file_name, ce)
         return pd.DataFrame()
     except pd.errors.EmptyDataError as ede:
-        logging.warning("Downloaded file %s is empty: %s", file_name, ede)
+        logging.warning("Downloaded file %s is empty: %s", tmp_file_name, ede)
         return pd.DataFrame()
     except Exception as e:
         logging.error(
-            "Unexpected error while downloading %s from S3: %s", file_name, e)
+            "Unexpected error while downloading %s from S3: %s", tmp_file_name, e)
         raise
```

Comment:
why is the exception the only one not returning a dataframe? should they all raise their own error to interrupt the flow?

Reply:
For some of these errors continuing with an empty dataframe is acceptable, however others represent failures that need explicit handling.